### PR TITLE
A compatibility test against the ocaml-mustache v2.0.0 API

### PR DESCRIPTION
**Pull request by gasche**
This test is designed to get a sense of the backward-compatibility
impact of changes to the ocaml-mustache library.

mustache_v200.mli is exactly a copy of the Mustache interface as it
existed in the version v2.0.0, and it should not change.

mustache_v200.ml is a reimplementation of this interface using the
current ocaml-mustache library. If the library changes, this
reimplementation will have to be fixed as well; the invasiveness of
this fix can be used to estimate the invasiveness of the change for
end-users.

v2.0.0 was chosen because it does not contain an explicit AST
definition or "fold" function -- whose compatibility breaks for
basically any new language feature or representation
change (for example when adding locations in the AST).

### CODE REVIEW

File: lib_test/compat/dune [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(tests
+ (libraries mustache ezjsonm)
+ (names user_program)
+ (deps user_program.exe))
```
Comment (rgrinberg) [inline lib_test/compat/dune:4]: Dune already adds the dependency on the binary automatically.

File: lib_test/compat/user_program.expected [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
+# Parsed template
+
+## parsed
```
Comment (rgrinberg) [inline lib_test/compat/user_program.expected:4]: I don't see this file being used anywhere in the test.

File: lib_test/compat/dune (hunk 1 above)
Comment (gasche) [reply lib_test/compat/dune:4]: I removed the `(deps ..)` stanza, thanks.

File: lib_test/compat/user_program.expected (hunk 2 above)
Comment (gasche) [reply lib_test/compat/user_program.expected:4]: `dune` compares this output with the output generated by running `user_program.exe` and fails if they differ.

**Comment by gasche** [comment]
Regarding backwards-compatibility:

- I expect that several users of Mustache are using the library for one-off scripts that they do not publish on opam. Taking myself as an example, I want to use ocaml-mustache for (1) a low-key website generator and (2) ugly scripts to generate emails to people. I don't think that any of these projects would end up on opam. There is more contributor activity in the project than the dependencies suggest, which is consistent with my imagination. (Note: there are now [3 dependents](https://opam.ocaml.org/packages/mustache/), with [camyll](https://opam.ocaml.org/packages/camyll/) released last October.)

- The idea of this test is not to enforce backward-compatibility forever (I agree with you that it would be silly), but to get some idea of the "amount of breakage" when we change things. I feel more confident working on a codebase when I have some way to say: well this change is probably minor, but this one will affect potential users.
